modules/Bio/EnsEMBL/Compara/RunnableDB/LoadMLSSids.pm

```diff
@@ -0,0 +1,178 @@
+
+=item method_type
+
+Mandatory. Method type.
+
+=item species_set_name
+
+Mandatory. Species set name.
+
+=item release
+
+Mandatory. Release to use as current.
+
+=item add_prev_mlss
+
+Optional. Also fetch the previous MLSS id of the same method type and species
+set name and return it under "prev_mlss_id" key. By default, do not fetch it.
+
+=item add_sister_mlsss
+
+Optional. Also fetch the sister MLSS ids of the main MLSS:
+ - If method_type is EPO, fetch the EPO Extended MLSS id and its related GERP
+   MLSS ids and return them under "ext_mlss_id", "ce_mlss_id" and "cs_mlss_id"
+   keys, respectively
+ - If method_type is EPO_EXTENDED or PECAN, fetch the GERP MLSS ids and return
+   them under "ce_mlss_id" and "cs_mlss_id" keys
+By default, do not fetch them.
+
+=item branch_code
+
+Optional. Flow the MLSS id(s) into the given branch instead of saving them as
+pipeline-wide parameters. By default, save the MLSS id(s) as pipeline-wide
+parameters.
+
+=back
+
+=head1 EXAMPLES
+
+    standaloneJob.pl Bio::EnsEMBL::Compara::RunnableDB::LoadMLSSids \
+        --reg_conf $COMPARA_REG_PATH --master_db compara_master \
+        --method_type EPO --species_set_name mammals --release $CURR_ENSEMBL_RELEASE
+
+    standaloneJob.pl Bio::EnsEMBL::Compara::RunnableDB::LoadMLSSids \
+        --master_db $(cp1 details url ensembl_compara_master) \
+        --method_type EPO --species_set_name sauropsids --release 101 \
+        --add_prev_mlss 1 --add_sister_mlsss 1 --branch_code 2
+
+=cut
+
+package Bio::EnsEMBL::Compara::RunnableDB::LoadMLSSids;
+
+use warnings;
+use strict;
+
+use Bio::EnsEMBL::Registry;
+use Bio::EnsEMBL::Compara::DBSQL::DBAdaptor;
+
+use base ('Bio::EnsEMBL::Compara::RunnableDB::BaseRunnable');
+
+
+sub param_defaults {
+    my $self = shift;
+
+    return {
+        %{$self->SUPER::param_defaults},
+
+        'add_prev_mlss'     => 0,
+        'add_sister_mlsss'  => 0,
+        'branch_code'       => undef,
+    }
+}
+
+
+sub fetch_input {
+    my $self = shift;
+    my $master_dba = $self->get_cached_compara_dba('master_db');
+    my $release = $self->param_required('release');
+    my $method_type = $self->param_required('method_type');
+    my $species_set_name = $self->param_required('species_set_name');
+
+    my %mlss_ids;
+    my $mlss_adaptor = $master_dba->get_MethodLinkSpeciesSetAdaptor;
+    # First, fetch all the MLSSs that match the given method type and species set name
+    my @mlsss = grep { $_->species_set->name eq $species_set_name }
+        @{ $mlss_adaptor->fetch_all_by_method_link_type($method_type) };
+    $self->throw("No MLSSs found for method '$method_type' and species set '$species_set_name'") unless @mlsss;
+
+    # Fetch the "current" MLSS id (for the given release number)
+    my @curr_mlss = grep { $_->is_in_release($release) } @mlsss;
+    $self->throw("No MLSS found for method '$method_type' and species set '$species_set_name' in release $release") unless @curr_mlss;
+    my $mlss = $curr_mlss[0];
```

Comment:
What if `@curr_mlss` has more than 1 element ?

Reply:
I thought for a given method type and species set name we can only have 1 MLSS in the current release? I expected that something before would have complained about a malformed master database if not.

Reply:
There is no constraint in the code / schema to enforce that, it is just a convention. But you're right, there should be one at most in any given release.
I thought we had a HC/DC to check that in, but I can't find it. @CristiGuijarro : do you remember seeing one ? If not, we should create a Jira ticket to add this check

Reply:
There isn't a DC for that yet, but an appropriate place to put it might be in `CheckMSANames.pm`.
When I was working on the rice `species_sets` for the rice MSAs I was able to name every `species_set` "rice", and use those different `species_sets` with the **same** names for multiple attempts of the same `EPO` method. So this wasn't in a release db - but this sort of scenario didn't occur to me at the time as a new recruit.

Reply:
We need this in release dbs as that's the assumption behind `fetch_by_method_link_type_species_set_name`. That's also advertised on the page of each multiple alignment, e.g. http://www.ensembl.org/info/genome/compara/mlss.html?mlss=183

In master dbs, it's probably generally the case for every release number ever, but there could be a couple of failures here and there. Also, like you said it's useful to break the rule when doing some tests. I also tested EPO with different `fish` species-sets end of last year. But I used my own copy of the master db for the duration of the tests (so did you, if I remember correctly), and I was not running HCs/DCs on it. So, for the master db, I guess it's up to us to decide if and how we want to enforce it :wink: but the same check as in the release db should work ?